service/lib/agama/software/manager.rb

```diff
@@ -747,6 +750,41 @@ def copy_zypp_to_target
         FileUtils.copy(glob_credentials, target_dir)
       end
 
+      # private class to ensure that cfa reads installed system
+      # YaST target file does not work reliably as Agama does not have
+      # always switched SCR
+      class TargetFile
+        # Reads file content with respect of changed root in installation.
+        def self.read(path)
+          ::File.read(final_path(path))
+        end
+
+        # Writes file content with respect of changed root in installation.
+        def self.write(path, content)
+          ::File.write(final_path(path), content)
+        end
+
+        def self.final_path(path)
+          ::File.join("/mnt", path)
+        end
+        private_class_method :final_path
+      end
+
+      def modify_zypp_conf
+        # use defaults unless user explicitelly wants only required packages
+        return unless proposal.only_required
+
+        zypp_conf = Yast::Packager::CFA::ZyppConf.new(file_handler: TargetFile)
+        zypp_conf.load
+        tree = zypp_conf.generic_get("main")
+        if !tree
+          tree = ::CFA::AugeasTree.new
+          zypp_conf.generic_get("main", tree)
+        end
+        zypp_conf.generic_set("solver.onlyRequires", "true", tree)
```

Comment:
```suggestion
        zypp_conf.generic_set("solver.onlyRequires", "#{!!proposal.only_required}", tree)
```

This is only slightly more complex and guards against stupid mistakes when the default changes